src/test/java/org/openrewrite/javascript/tree/TupleTest.java

```diff
@@ -59,4 +56,44 @@ void typed() {
           )
         );
     }
+
+    @ExpectedToFail("Requires spread operator")
+    @Test
+    void spreadOperators() {
+        rewriteRun(
+          javaScript(
+            """
+              function concat(arr1, arr2) {
+                  return [...arr1, ...arr2]
+              }
+              """
+          )
+        );
+    }
+
+    @ExpectedToFail("Requires ArrayBindingPattern.")
+    @Test
+    void arrayBindingPattern() {
+        rewriteRun(
+          javaScript(
+            """
+              function tail(arg) {
+                  const [_, ...result] = arg;
+                  return result;
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void trailingCommas() {
```

Comment:
Let's also add a test for [named tuple members](https://www.typescriptlang.org/play?ts=4.0.2#example/named-tuples). Were you planning on addressing those separately? If so, let's create an issue for that.

Reply:
Ah, thanks for pointing that out!

I used the TS documentation for [reference](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html) and thought it was covered by [typed](https://github.com/openrewrite/rewrite-javascript/blob/d045ad8192c0f37632eb062177239efb078aa83d/src/test/java/org/openrewrite/javascript/tree/TupleTest.java#L50), but it wasn't -- updated the test!